File: app/src/main/java/com/ed/redditapp/ui/subreddit/FirstItemOffsetDecoration.java
package com.ed.redditapp.ui.subreddit;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class FirstItemOffsetDecoration extends RecyclerView.ItemDecoration {
    private final int offset;

    public FirstItemOffsetDecoration(int offset) {
        this.offset = offset;
    }

    @Override
    public void getItemOffsets(
            @NonNull Rect outRect,
            @NonNull View view,
            RecyclerView parent,
            @NonNull RecyclerView.State state) {
        if (parent.getChildAdapterPosition(view) == 0) {
            outRect.top = offset;
        }
    }
}

File: app/src/main/java/com/ed/redditapp/ui/subreddit/FirstItemOffsetDecoration.kt
package com.ed.redditapp.ui.subreddit

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration

class FirstItemOffsetDecoration(private val offset: Int) : ItemDecoration() {
    override fun getItemOffsets(
            outRect: Rect,
            view: View,
            parent: RecyclerView,
            state: RecyclerView.State) {
        if (parent.getChildAdapterPosition(view) == 0) {
            outRect.top = offset
        }
    }
}